src/updateComponentNode.js

```diff
@@ -158,15 +158,15 @@ export default function updateComponentNode (part, node, oldNode, context, force
        */
 
     if (componentInstance instanceof PureComponent) {
-      shouldUpdate = !shallowEqual(state, prevState) || !shallowEqual(props, prevProps);
+      shouldUpdate = !shallowEqual(state, prevState) || !shallowEqual(props, prevProps) || forceRender;
```

Comment:
instead of `|| forceRender` add it on if condition so we can avoid shallowCheck.
```js
if (componentInstance instanceof PureComponent && !forceRender) {
```